lib/src/types.rs

```diff
@@ -59,15 +59,23 @@ impl FieldDecl {
     }
 }
 
-#[derive(Debug, Clone, Deserialize, Serialize)]
+#[derive(Debug, Clone, Deserialize, Serialize, PartialEq)]
 pub struct MethodDecl {
     pub ret_type: Type,
     pub name: String,
     pub params: Vec<(Type, String)>,
     pub body: Stmt,
 }
 
-#[derive(Debug, Clone, Deserialize, Serialize)]
+impl MethodDecl {
+    pub fn as_bytes(&self, constant_pool: &mut ConstantPool) -> Vec<u8> {
+        let mut bytes = Vec::new();
+        // TODO
+        bytes
+    }
+}
+
+#[derive(Debug, Clone, Deserialize, Serialize, PartialEq)]
 pub enum Stmt {
```

Comment:
Come to think of it: Why is LovalVarDecl defined as 
```rs
    LocalVarDecl(Type, String), // first type of the local variable, then it's name
```
codegen needs an expression to assign to the LocalVarPool here

Reply:
Nevermind i forgot it's possible to do this:
```rs
i = true;
```